src/libreminesgameengine.cpp

```diff
@@ -586,15 +586,20 @@ void LibreMinesGameEngine::SLOT_addOrRemoveFlag(const uchar _X, const uchar _Y)
     if(!principalMatrix[_X][_Y].isHidden)
         return;
 
-    if(principalMatrix[_X][_Y].hasFlag)
+    if(principalMatrix[_X][_Y].FlagType == 2)
     {
-        principalMatrix[_X][_Y].hasFlag = false;
-        iMinesLeft++;
+        principalMatrix[_X][_Y].FlagType = 0;
         Q_EMIT SIGNAL_unflagCell(_X, _Y);
     }
+    else if (principalMatrix[_X][_Y].FlagType == 1)
```

Comment:
It would be nice to have an boolean attribute of the GameEngine Class that indicates whether to use the question or not. This attribute would change a little this function.

This option would be set by the user with a check box on the preferences dialog.